File: Data Structures & Algorithms/duplicate-integer/submission-4.java
class Solution 
{
    public boolean hasDuplicate(int[] nums) 
    {
        HashMap<Integer, Integer> x = new HashMap<Integer, Integer>();

        for(int i = 0; i < nums.length; i++)
        {
            if(x.containsKey(nums[i])){
                return true;
            }
            else{
                x.put(nums[i], 2);
            }
        }
        
        return false;
    }
}

File: Data Structures & Algorithms/binary-search/submission-4.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:

        p1 = 0
        p2 = len(nums) - 1

        while p1 <= p2:
            split = p1 + (p2-p1)//2

            if nums[split] > target:
                p2 = split - 1
            elif nums[split] < target:
                p1 = split + 1
            else: 
                return split
        return -1

